include/openenclave/bits/eeid.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file eeid.h
+ *
+ * This file defines the EEID structure.
+ *
+ */
+
+#ifndef _OE_BITS_EEID_H
+#define _OE_BITS_EEID_H
+
+#include "properties.h"
+
+OE_EXTERNC_BEGIN
+
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+/*
+**==============================================================================
+**
+** oe_eeid_t
+**
+**==============================================================================
+*/
+typedef struct oe_eeid_t_
+{
+    struct
+    {
+        uint32_t H[8];
+        uint32_t N[2];
+    } hash_state; /* internal state of the hash computation at the end of
+                           the enclave base image */
+    uint64_t signature_size; /* size of signature */
+    uint8_t signature[1808]; /* base-image signature and associated data (for
+                           SGX, the complete sigstruct of the base image) */
+    oe_enclave_size_settings_t
```

Comment:
In this structure, I'd like to see something like `ConfigID` to align with Icelake. It could be anything that uniquely identifies the enclave instance. In the case of SGX-LKL, it could be the root hash of the application. The verifier then just have to look at this data structure in most cases, and additionally verify the custom `data` section only when necessary. For that, we need the upper layer that constructed the `data` section to provide a customizable verifier.

Reply:
If we move the application root hash into `data[]`, as we discussed offline, the data section should be application specific,  so the changes to SGX-LKL doesn't affect its verification. Make sense?

Reply:
Sorry, doesn't make sense. What's the "application" here? The EEID data is instance-specific, i.e. specific to a particular enclave instance. If that instance happens to be an SGX-LKL enclave, then it is specific to SGX-LKL. In our current plans for SGX-LKL, the EEID data is a JSON file that contains the entire SGX-LKL configuration, including command line, environment variables, file system hashes, names of keys; essentially everything that ever will need to be attested. (But for OE, this data is opaque, so none of that needs to go into this particular header.)

Reply:
As we discussed offline, the `application-specific` bits should include the command line, the file system hash, and a few environment variables that have security impacts. The relying party has no interests in some of the SGX-LKL specific environment settings such as how many ethread we are running, stack size, TAP MTU, etc.

Reply:
Agree this is not OE specific. Move the discussion to another venue.